gamification-github-services/src/main/java/org/exoplatform/gamification/github/services/impl/WebhookServiceImpl.java

```diff
@@ -0,0 +1,416 @@
+      return processRequest(httpClient, request);
+    } catch (IOException e) {
+      throw new GithubConnectionException(GITHUB_CONNECTION_ERROR, e);
+    }
+  }
+
+  private String processPost(URI uri, String jsonString, String accessToken) throws GithubConnectionException {
+    HttpClient httpClient = getHttpClient();
+    HttpPost request = new HttpPost(uri);
+    StringEntity entity = new StringEntity(jsonString, ContentType.APPLICATION_JSON);
+    try {
+      request.setHeader(HTTP.CONTENT_TYPE, MediaType.APPLICATION_JSON);
+      request.setHeader(AUTHORIZATION, TOKEN + accessToken);
+      request.setEntity(entity);
+      return processRequest(httpClient, request);
+    } catch (IOException e) {
+      throw new GithubConnectionException(GITHUB_CONNECTION_ERROR, e);
+    }
+  }
+
+  private String processDelete(URI uri, String accessToken) throws GithubConnectionException {
+    HttpClient httpClient = getHttpClient();
+    HttpDelete request = new HttpDelete(uri);
+    try {
+      request.setHeader(AUTHORIZATION, TOKEN + accessToken);
+      return processRequest(httpClient, request);
+    } catch (IOException e) {
+      throw new GithubConnectionException(GITHUB_CONNECTION_ERROR, e);
+    }
+  }
+
+  private String processRequest(HttpClient httpClient, HttpRequestBase request) throws IOException, GithubConnectionException {
+    HttpResponse response = httpClient.execute(request);
+    boolean isSuccess = response != null
+        && (response.getStatusLine().getStatusCode() >= 200 && response.getStatusLine().getStatusCode() < 300);
+    if (isSuccess) {
+      return processSuccessResponse(response);
+    } else if (response != null && response.getStatusLine().getStatusCode() == 404) {
+      return null;
+    } else {
+      processErrorResponse(response);
+      return null;
+    }
+  }
+
+  private String processSuccessResponse(HttpResponse response) throws IOException {
+    if (response.getStatusLine().getStatusCode() == HttpStatus.SC_NO_CONTENT) {
+      return String.valueOf(HttpStatus.SC_NO_CONTENT);
+    } else if ((response.getStatusLine().getStatusCode() == HttpStatus.SC_CREATED
+        || response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) && response.getEntity() != null
+        && response.getEntity().getContentLength() != 0) {
+      try (InputStream is = response.getEntity().getContent()) {
+        return IOUtils.toString(is, StandardCharsets.UTF_8);
+      }
+    } else {
+      return null;
+    }
+  }
+
+  private void processErrorResponse(HttpResponse response) throws GithubConnectionException, IOException {
+    if (response == null) {
+      throw new GithubConnectionException("Error when connecting github");
+    } else if (response.getEntity() != null) {
+      try (InputStream is = response.getEntity().getContent()) {
+        String errorMessage = IOUtils.toString(is, StandardCharsets.UTF_8);
+        if (StringUtils.contains(errorMessage, "wom.")) {
```

Comment:
To verify the Response parsing algorithm

```suggestion
        if (StringUtils.contains(errorMessage, "")) {
```